src/iceberg_manifest.cpp

```diff
@@ -306,7 +421,8 @@ idx_t IcebergManifestEntryV2::ProduceEntries(DataChunk &chunk, idx_t offset, idx
 				entry.sequence_number = sequence_numbers[index];
 			} else {
 				//! Value should only be NULL for ADDED manifest entries, to support inheritance
-				D_ASSERT(entry.status == IcebergManifestEntryStatusType::ADDED);
+				//! FIXME: the 'persistent' tables have DELETED entries with no sequence number, while the metadata says
+				//! v2 According to the iceberg spec, those tables are corrupt
```

Comment:
This is really odd, and should be looked into
the `sequence_number` is allowed to be NULL for an entry with the status `ADDED`, so it can inherit from the manifest.
Once it's not ADDED, it should have a sequence number, to indicate that it was created by that sequence

Reply:
perhaps this was just written incorrectly by spark at the time this data was generated? we could try to regen the data

Reply:
Made #215 for this 👍